src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -102,8 +102,17 @@ export function usePanZoom(): ReactThreeFiber.Events {
       // eslint-disable-next-line no-param-reassign
       camera.zoom = Math.max(1, camera.zoom * factor);
       camera.updateProjectionMatrix();
+
+      const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
+      const { x: pointX, y: pointY } = projectedPoint;
+      const { x: camX, y: camY } = camera.position;
+
+      moveCamera(
+        camX + pointX * (1 - 1 / factor),
+        camY + pointY * (1 - 1 / factor)
```

Comment:
Pardon my non-scientific notations...

`point - point / factor` <=> `A - B`, where:
- `A` is the point in camera space where the cursor is located before the zoom,
- `B` is where that point moves to once the zoom is applied.

Reply:
I just move the camera by that amount. `moveCamera` then takes care of the clamping.

Reply:
In addition, passing a `Vector` as arg to `moveCamera` rather than the two coordinates could avoid destructuring in both `onPointerMove` and `onWheel`.

Reply:
- `moveCameraTo` 👍 
- I used `point` for consistency with `evt.unprojectedPoint` - it refers to the position of the pointer (which could be the mouse cursor or a finger on touchscreens).
- I'd also prefer to work with vectors but:
  - I don't want to move the camera along `z`, so it doesn't really make sense to work with and pass a `Vector3`, and I can't find a simple method to convert a `Vector3` to a `Vector2`.
  - Three.JS's vector API is a mutable API, so I have to clone the vectors before performing any operations on them, which affects readability and is prone to bugs. I feel like destructuring brings more clarity overall.

Reply:
`point` in unfortunately extremely generic... `pointer` ?

Yeah, after seeing the API of `Vector`, I feel your solution is clearer.